EIPS/eip-6239.md

```diff
@@ -0,0 +1,323 @@
+
+For example:
+
+```text
+@prefix : <http://example.org/entity/> .
+@prefix p: <http://example.org/property/> .
+
+<http://example.org/graphs/example>
+  {
+       :user1 a :User;
+              p:name ”Alice” .
+
+   }
+```
+
+In the contract events: `CreateRDF`, `UpdateRDF`, `RemoveRDF`, and the `rdfOf method`, the  `rdfStatements` is used in `ttl` format by default. If other formats listed above are used, a format identifier needs to be added for identification.
+
+The format identifier starts with `[` and ends with `]` with the format in the middle, i.e., `[format]`.
+
+For example, the `rdfStatements` in `nt` format should include the prefix `[nt]`.
+
+```text
+[nt]subject predicate object .
+```
+
+
+### Contract Interface
+
+```solidity
+/**
+ * @title Semantic Soulbound Token
+ * Note: the ERC-165 identifier for this interface is 0xfbafb698
+ */
+interface ISemanticSBT{
+    /**
+     * @dev This emits when minting a Semantic Soulbound Token.
+     * @param tokenId The identifier for the Semantic Soulbound Token.
+     * @param rdfStatements The RDF statements for the Semantic Soulbound Token. 
+     */
+    event CreateRDF (
+        uint256 indexed tokenId,
+        string  rdfStatements
+    );
+    /**
+     * @dev This emits when updating the RDF data of Semantic Soulbound Token. RDF data is a collection of RDF statements that are used to represent information about resources.
+     * @param tokenId The identifier for the Semantic Soulbound Token.
+     * @param rdfStatements The RDF statements for the semantic soulbound token. 
+     */
+    event UpdateRDF (
+        uint256 indexed tokenId,
+        string  rdfStatements
+    );
+    /**
+     * @dev This emits when burning or revoking Semantic Soulbound Token.
+     * @param tokenId The identifier for the Semantic Soulbound Token.
+     * @param rdfStatements The RDF statements for the Semantic Soulbound Token. 
+     */
+    event RemoveRDF (
+        uint256 indexed tokenId,
+        string  rdfStatements
+    );
+    /**
+     * @dev Returns the RDF statements of the Semantic Soulbound Token. 
+     * @param tokenId The identifier for the Semantic Soulbound Token.
+     * @return rdfStatements The RDF statements for the Semantic Soulbound Token. 
+     */
+    function rdfOf(uint256 tokenId) external view returns (string memory rdfStatements);
+}
+```
+
+`ISemanticRDFSchema`, an extension of ERC-721 Metadata, is **OPTIONAL** for this standard, it is used to get the Schema URI for the RDF data.
+
+```solidity
+interface ISemanticRDFSchema{
+    /**
+     * @notice Get the URI of schema for this contract.
+     * @return The URI of the contract which point to a configuration profile.
+     */
+    function schemaURI() external view returns (string memory);
+}
+```
+
+`ISemanticSBTUpdate` is an extension interface, that is **OPTIONAL** for this standard, used to update the RDF data for the Semantic Soulbound Token.
```

Comment:
Generally speaking, functions that are called by the deployer of the contract don't need to be standardized, because the deployer of the contract has application specific knowledge about the implementation. For example the `mint` and `burn` functions of ERC-20 and ERC-721 tokens are part of those standards because that behaviour is specific to each implementation.

Similar logic applies here. `ISemanticSBTUpdate` is specific to each implementation, and so shouldn't be part of the standard.

Reply:
@SamWilsn Thank you for your suggestion, we have deleted ISemanticSBTUpdate in eip 6239.